grails-app/controllers/com/unifina/controller/api/PermissionApiController.groovy

```diff
@@ -216,10 +218,32 @@ class PermissionApiController {
 
 	@StreamrApi(authenticationLevel = AuthLevel.NONE)
 	def delete(String id) {
-		usePermission(params.resourceClass, params.resourceId, id as Long) { p, res ->
-			// share-permission has been tested in usePermission (calls useResource)
-			permissionService.systemRevoke(p)
-			render status: 204
+		if (!params.resourceClass) {
+			throw new IllegalArgumentException("Missing resource class")
+		}
+		if (!grailsApplication.isDomainClass(params.resourceClass)) {
+			throw new IllegalArgumentException("${params.resourceClass.simpleName} is not a domain class!")
+		}
+		def resource = params.resourceClass.get(params.resourceId)
+		if (!resource) {
+			throw new NotFoundException(params.resourceClass.simpleName, params.resourceId.toString())
+		}
+		Permission permission = permissionService.getPermissionsTo(resource).find { it.id.toString() == id.toString() }
+		if (!permission) {
+			throw new NotFoundException("permissions", id.toString())
+		}
+		Userish user = request.apiUser ?: request.apiKey
+		if (permissionService.canShare(user, resource)) {
+			// user with share permission to resource can delete another user's permission to same resource
+			permissionService.systemRevoke(permission)
+			render(status: 204)
+		} else if (permission.user == user) {
+			// user without share permission to resource can delete their own permission to resource
+			permissionService.systemRevoke(permission)
+			render(status: 204)
+		} else {
+			// user without share permission to resource can't delete another user's permission to same resource
+			render(status: 403)
```

Comment:
`throw new NotPermittedException(...)` to give a proper error response to the user